YunoHost/yunohost
# [enh] Simplify the whole LDAP interface thing

## The problem

c.f. https://github.com/YunoHost/moulinette/pull/183

## Solution

c.f. https://github.com/YunoHost/moulinette/pull/183

This is a cherry-pick from https://github.com/YunoHost/yunohost/pull/585 for the part that handle LDAP authentication using root instead of asking for the admin password each time. Considering that the whole group permission PR is likely to take some time, yet this thing is a pretty interesting independent feature ...

I started iterating on it to also get rid of the whole '`auth`' everywhere madness and simplify other related thing in the actionmap (the mysterious 'authenticate' which nobody knows how to deal with anymore or what it does exactly)

So to summarize the two major benefits right now are :  
- **simpler / monre consistent UX** (no weird behavior for CLI user where some commands would ask you the admin password even tho you are already authenticated as root/admin)
- **simplify code by removing the need to pass the old `auth` object everywhere** (e.g. for every function where you would like to later call `domain_list` or `app_ssowatconf` in a sub-sub-sub-sub-call)


## PR Status

Made a quick test and it looks like this is a working, but considering this is a touchy part I want to do moar tests. Also I might find some fancy stuff to factorize a bit the part where we fetch the `auth` object.

## How to test

... Checkout this branch (and the corresponding one). You need to make sure that the ldap conf is updated otherwise this won't work ... then try any step that usually requires to enter the admin password (e.g. postinstall or user create)

## Validation
